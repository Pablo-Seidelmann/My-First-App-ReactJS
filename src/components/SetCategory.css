.div-list
{
    margin-bottom:  20px;
    text-align: center;
}

.list
{
    color: #5e5e5e;
    font-weight: bold;
}