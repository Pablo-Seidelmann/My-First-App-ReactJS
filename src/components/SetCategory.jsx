import React from 'react';
import PropTypes from 'prop-types'; 
import styled from 'styled-components';

const SetCategory = ({ category }) => {

    const Div = styled.div`
        text-align: center;
        margin-bottom:  20px;
    `

    const Li = styled.li`
        font-weight: bold;
        color: #5e5e5e;
    `

    return (

        <Div>
            
            <Li id = 'list-item'>
                { category }
            </Li>

        </Div>

    )
}

SetCategory.propTypes = {
    category: PropTypes.string.isRequired
}

export default SetCategory;
