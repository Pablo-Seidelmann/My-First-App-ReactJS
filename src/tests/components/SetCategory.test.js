import { shallow } from 'enzyme';
import React from 'react';
import SetCategory from '../../components/SetCategory'; 

describe('Pruebas a el Componente <SetCategory />', () => {
    
    const category = 'Hola Mundo';

    const wrapper = shallow( <SetCategory category = { category } /> );

    test('SnapShot al Componente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de haber una lista y dentro la category', () => {
        
        const listItem = wrapper.find('.list').text().trim();

        expect( listItem ).toBe( category );

    });
    
    

});
