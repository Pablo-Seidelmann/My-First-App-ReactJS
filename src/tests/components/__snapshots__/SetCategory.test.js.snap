// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a el Componente <SetCategory /> SnapShot al Componente 1`] = `
<styled.div>
  <styled.li
    id="list-item"
  >
    Hola Mundo
  </styled.li>
</styled.div>
`;
